### PR TITLE
JA translation user agent reduction snippets

### PR DESCRIPTION
**Pull request by mattletford**
@agektmr JA translation user agent reduction snippets

### CODE REVIEW

File: site/ja/docs/privacy-sandbox/user-agent/snippets/index.md [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+
+## 背景
+
+今後、Chrome の [User-Agent 文字列中で開示される情報が削減されていきます](/docs/privacy-sandbox/user-agent/)。これは、一定期間をかけて段階的に行われ、最終的には
+OS のバージョン、デバイス、ブラウザの詳細バージョンの部分が User-Agent
+文字列から削除されます。
+
+モバイル デバイスの場合の例を次に示します。
+
+現在
```
Comment (agektmr) [inline site/ja/docs/privacy-sandbox/user-agent/snippets/index.md:28]:
```suggestion
{% Compare 'worse', 'old' %}
```

File: site/ja/docs/privacy-sandbox/user-agent/snippets/index.md [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+
+`Mozilla/5.0 (Linux; Android 12; Pixel 5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/95.0.4638.16 Mobile Safari/537.36`
+
```
Comment (agektmr) [inline site/ja/docs/privacy-sandbox/user-agent/snippets/index.md:31]:
```suggestion
{% endCompare %}

```

File: site/ja/docs/privacy-sandbox/user-agent/snippets/index.md [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+情報量削減後
```
Comment (agektmr) [inline site/ja/docs/privacy-sandbox/user-agent/snippets/index.md:32]:
```suggestion
{% Compare 'better', 'new' %}
```

File: site/ja/docs/privacy-sandbox/user-agent/snippets/index.md [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+
+`Mozilla/5.0 (Linux; Android 10; K) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/95.0.0.0 Mobile Safari/537.36`
+
```
Comment (agektmr) [inline site/ja/docs/privacy-sandbox/user-agent/snippets/index.md:35]:
```suggestion
{% endCompare %}

```

File: site/ja/docs/privacy-sandbox/user-agent/snippets/index.md [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+クライアントサイドの JavaScript で [navigator.userAgent](https://developer.mozilla.org/docs/Web/API/Navigator/userAgent) の現在の値を情報量削減後の形式でオーバーライドするには、次のスニペットを使用します。
+
+デモ: [reduced-ua.glitch.me/javascript.html](https://reduced-ua.glitch.me/javascript.html)
+
+```js
+const chromeUAs =
+  /^Mozilla\/5\.0\(((?<platform>Lin|Win|Mac|X11; C|X11; L)+[^\)]+)\)AppleWebKit\/537.36 \(KHTML, like Gecko\)Chrome\/(?<major>\d+)[\d\.]+(?<mobile>[ Mobile]*)Safari\/537\.36$/;
+
+const matched = chromeUAs.exec(navigator.userAgent);
```
Comment (agektmr) [inline site/ja/docs/privacy-sandbox/user-agent/snippets/index.md:84]:
```suggestion
const chromeUAs = /^Mozilla\/5\.0\(((?<platform>Lin|Win|Mac|X11; C|X11; L)+[^\)]+)\)AppleWebKit\/537.36 \(KHTML, like Gecko\)Chrome\/(?<major>\d+)[\d\.]+(?<mobile>[ Mobile]*)Safari\/537\.36$/;
const matched = chromeUAs.exec(navigator.userAgent);
```

File: site/ja/docs/privacy-sandbox/user-agent/snippets/index.md [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+description: >
+  Test your own sites and services against the reduced Chrome user-agent format by using a regular expression to transform the current Chrome user-agent to the reduced format.
+date: 2021-11-26
+authors:
+  - rowan_m
+---
+
+現在の Chrome の User-Agent
+文字列を情報量削減後の形式に変換するためのコード スニペットです。
+
```
Comment (agektmr) [inline site/ja/docs/privacy-sandbox/user-agent/snippets/index.md:15]: Remove these three lines.

```suggestion
```

File: site/ja/docs/privacy-sandbox/user-agent/snippets/index.md [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+    'X11; C': 'X11; CrOS x86_64',
+    'X11; L': 'X11; Linux x86_64',
+  };
+
+  const reducedUA =
+    `Mozilla/5.0 (${unifiedPlatform[matched.groups.platform]}) ` +
+    `AppleWebKit/537.36 (KHTML, like Gecko) ` +
+    `Chrome/${matched.groups.major}.0.0.0${matched.groups.mobile} Safari/537.36`;
+
+  // navigator.userAgent を情報量削減後の文字列でオーバーライドする
```
Comment (agektmr) [inline site/ja/docs/privacy-sandbox/user-agent/snippets/index.md:101]:
```suggestion
  const reducedUA =
        `Mozilla/5.0 (${unifiedPlatform[matched.groups.platform]}) ` +
        `AppleWebKit/537.36 (KHTML, like Gecko) ` +
        `Chrome/${matched.groups.major}.0.0.0${matched.groups.mobile} Safari/537.36`;
  // navigator.userAgent を情報量削減後の文字列でオーバーライドする
```

File: site/ja/docs/privacy-sandbox/user-agent/snippets/index.md [hunk 8]
```diff
@@ -0,0 +1,176 @@
...
+### Node.js
+
+[Node.js](https://nodejs.org/) で受信した [User-Agent ヘッダー](https://developer.mozilla.org/docs/Web/HTTP/Headers/User-Agent)を情報量削減後の形式でオーバーライドするには、次のスニペットを使用します。
+
+デモ: [reduced-ua.glitch.me/server-side](https://reduced-ua.glitch.me/server-side)
+
+```js
+const chromeUAs = /^Mozilla\/5\.0 \(((?<platform>Lin|Win|Mac|X11; C|X11; L)+[^\)]+)\) AppleWebKit\/537.36 \(KHTML, like Gecko\)Chrome\/(?<major>\d+)[\d\.]+(?<mobile>[ Mobile]*) Safari\/537\.36$/;
+
+const matched = chromeUAs.exec(request.get(\'user-agent\'));
```
Comment (agektmr) [inline site/ja/docs/privacy-sandbox/user-agent/snippets/index.md:119]:
```suggestion
const chromeUAs = /^Mozilla\/5\.0 \(((?<platform>Lin|Win|Mac|X11; C|X11; L)+[^\)]+)\) AppleWebKit\/537.36 \(KHTML, like Gecko\)Chrome\/(?<major>\d+)[\d\.]+(?<mobile>[ Mobile]*) Safari\/537\.36$/;
const matched = chromeUAs.exec(request.get(\'user-agent\'));
```

File: site/ja/docs/privacy-sandbox/user-agent/snippets/index.md [hunk 9]
```diff
@@ -0,0 +1,176 @@
...
+'Lin': 'Linux; Android 10; K',
+'Win': 'Windows NT 10.0; Win64; x64',
+'Mac': 'Macintosh; Intel Mac OS X 10_15_7',
+'X11; C': 'X11; CrOS x86_64',
+'X11; L': 'X11; Linux x86_64',
+};
+
+request.headers['user-agent'] = `Mozilla/5.0 (${unifiedPlatform[matched.groups.platform]}) AppleWebKit/537.36(KHTML, like Gecko)Chrome/${matched.groups.major}.0.0.0${matched.groups.mobile}Safari/537.36`;
+
+}
```
Comment (agektmr) [inline site/ja/docs/privacy-sandbox/user-agent/snippets/index.md:133]:
```suggestion
if (matched) {
  const unifiedPlatform = {
    'Lin': 'Linux; Android 10; K',
    'Win': 'Windows NT 10.0; Win64; x64',
    'Mac': 'Macintosh; Intel Mac OS X 10_15_7',
    'X11; C': 'X11; CrOS x86_64',
    'X11; L': 'X11; Linux x86_64',
  };
  request.headers['user-agent'] = `Mozilla/5.0 (${unifiedPlatform[matched.groups.platform]}) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/${matched.groups.major}.0.0.0${matched.groups.mobile} Safari/537.36`;
}
```

File: site/ja/docs/privacy-sandbox/user-agent/snippets/index.md [hunk 10]
```diff
@@ -0,0 +1,176 @@
...
+        unifiedPlatform[matched.groups.platform]
+      }) AppleWebKit/537.36(KHTML, like Gecko)Chrome/${
+        matched.groups.major
+      }.0.0.0${matched.groups.mobile}Safari/537.36`
+    );
+    return await fetch(clonedRequest);
+  } else {
+    return await fetch(request);
+  }
+}
```
Comment (agektmr) [inline site/ja/docs/privacy-sandbox/user-agent/snippets/index.md:173]:
```suggestion
addEventListener('fetch', event => {
  event.respondWith(handleRequest(event.request));
})

async function handleRequest(request) {
  const chromeUAs = /^Mozilla\/5\.0 \(((?<platform>Lin|Win|Mac|X11; C|X11; L)+[^\)]+)\) AppleWebKit\/537.36 \(KHTML, like Gecko\) Chrome\/(?<major>\d+)[\d\.]+(?<mobile>[ Mobile]*) Safari\/537\.36$/;
  const matched = chromeUAs.exec(request.headers.get('user-agent'));

  if (matched) {
    const unifiedPlatform = {
      'Lin': 'Linux; Android 10; K',
      'Win': 'Windows NT 10.0; Win64; x64',
      'Mac': 'Macintosh; Intel Mac OS X 10_15_7',
      'X11; C': 'X11; CrOS x86_64',
      'X11; L': 'X11; Linux x86_64',
    };

    const clonedRequest = new Request(request, {referrer: request.referrer});
    clonedRequest.headers.set('user-agent', `Mozilla/5.0 (${unifiedPlatform[matched.groups.platform]}) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/${matched.groups.major}.0.0.0${matched.groups.mobile} Safari/537.36`);
    return await fetch(clonedRequest);
  } else {
    return await fetch(request);
  }
}
```

**Review by agektmr — CHANGES_REQUESTED** [review]

File: site/ja/docs/privacy-sandbox/user-agent/snippets/index.md [hunk 11]
```diff
@@ -0,0 +1,176 @@
+---
+layout: 'layouts/doc-post.njk'
+title: User-Agent の情報量削減スニペット
+subhead: >
+  Code snippets to transform the current Chrome user-agent to the reduced format.
+description: >
+  Test your own sites and services against the reduced Chrome user-agent format by using a regular expression to transform the current Chrome user-agent to the reduced format.
```
Comment (agektmr) [inline site/ja/docs/privacy-sandbox/user-agent/snippets/index.md:7]:
```suggestion
  現在の Chrome の User-Agent 文字列を、正規表現を使って情報量削減後の形式に変換することで、あなたのサイトやサービスが情報量削減後の形式で正常に動作するかをテストしてください。
```

**Review by agektmr — CHANGES_REQUESTED** [review]

File: site/ja/docs/privacy-sandbox/user-agent/snippets/index.md [hunk 12]
```diff
@@ -0,0 +1,176 @@
...
+
+この式から次の値が得られます。
+
+- `platform`: プラットフォームまたはオペレーティング システム
+- `major`: ブラウザのメジャー バージョン
+- `mobile`: モバイル デバイス インジケーター
+
+ここで得られた値を、次に示す情報量削減後の形式のテンプレート文字列に挿入します。
+
+`Mozilla/5.0 (\${unifiedPlatform\[matched.platform\]}) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/\${matched.major}.0.0.0\${matched.mobile} Safari/537.36`
```
[the remark this answers is not in the document]
Comment (mattletford) [reply site/ja/docs/privacy-sandbox/user-agent/snippets/index.md]: @agektmr I've added the code block with back-ticks

File: site/ja/docs/privacy-sandbox/user-agent/snippets/index.md [hunk 13]
```diff
@@ -0,0 +1,170 @@
...
+
+今後、Chrome の [User-Agent 文字列中で開示される情報が削減されていきます](/docs/privacy-sandbox/user-agent/)。これは、一定期間をかけて段階的に行われ、最終的には
+OS のバージョン、デバイス、ブラウザの詳細バージョンの部分が User-Agent
+文字列から削除されます。
+
+モバイル デバイスの場合の例を次に示します。
+
+{% Compare 'worse', 'old' %}
+
+`Mozilla/5.0 (Linux; Android 12; Pixel 5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/95.0.4638.16 Mobile Safari/537.36`
```
Comment (agektmr) [inline site/ja/docs/privacy-sandbox/user-agent/snippets/index.md:27]:
```suggestion
<span style="font-family: monospace">Mozilla/5.0
(Linux; Android <span style="background: #ef9a9a">12; Pixel 5</span>)
AppleWebKit/537.36 (KHTML, like Gecko) Chrome/95.<span  style="background:
#ef9a9a">0.4638.16</span> Mobile Safari/537.36</span>
```

File: site/ja/docs/privacy-sandbox/user-agent/snippets/index.md [hunk 14]
```diff
@@ -0,0 +1,170 @@
...
+情報量削減後の形式の `User-Agent` 文字列を使用できれば、それを自分のコードに照らして変換したりテストしたりできます（オーバーライドや置き換えを行う、新しいバージョンを作成して並列にテストするなど）。
+
+### 正規表現
+
+確実に変換可能な Chrome の User-Agent **のみ**に一致し、必要な変数値（後で新しい形式のテンプレートに挿入する）を抜き出すための正規表現です。
+
+コピーしやすいように 1 行で記述されています。長いですが、Chrome
+のみに一致させることと比較的低負荷なチェックで済ませることを優先した結果です。
+
+`/\^Mozilla\\/5\\.0 \\(((?Lin\|Win\|Mac\|X11; C\|X11; L)+\[\^\\)\]+)\\) AppleWebKit\\/537.36 \\(KHTML, like Gecko\\) Chrome\\/(?\\d+)\[\\d\\.\]+(?\[ Mobile\]\*) Safari\\/537\\.36\$/`
```
Comment (agektmr) [inline site/ja/docs/privacy-sandbox/user-agent/snippets/index.md:53]:
```suggestion
```text
/^Mozilla\/5\.0 \(((?<platform>Lin|Win|Mac|X11; C|X11; L)+[^\)]+)\) AppleWebKit\/537.36 \(KHTML, like Gecko\) Chrome\/(?<major>\d+)[\d\.]+(?<mobile>[ Mobile]*) Safari\/537\.36$/
```
```

File: site/ja/docs/privacy-sandbox/user-agent/snippets/index.md [hunk 15]
```diff
@@ -0,0 +1,170 @@
...
+    `AppleWebKit/537.36 (KHTML, like Gecko) ` +
+    `Chrome/${matched.groups.major}.0.0.0${matched.groups.mobile} Safari/537.36`;
+  // navigator.userAgent を情報量削減後の文字列でオーバーライドする
+  Object.defineProperty(navigator, 'userAgent', {
+    value: reducedUA,
+    writable: false,
+    configurable: true,
+  });
+}
+```
```
Comment (agektmr) [inline site/ja/docs/privacy-sandbox/user-agent/snippets/index.md:107]: Please replicate the original English version as much as possible (except the Japanese translation).

File: site/ja/docs/privacy-sandbox/user-agent/snippets/index.md [hunk 16]
```diff
@@ -0,0 +1,170 @@
...
+
+### Node.js
+
+[Node.js](https://nodejs.org/) で受信した [User-Agent ヘッダー](https://developer.mozilla.org/docs/Web/HTTP/Headers/User-Agent)を情報量削減後の形式でオーバーライドするには、次のスニペットを使用します。
+
+デモ: [reduced-ua.glitch.me/server-side](https://reduced-ua.glitch.me/server-side)
+
+```js
+const chromeUAs = /^Mozilla\/5\.0 \(((?<platform>Lin|Win|Mac|X11; C|X11; L)+[^\)]+)\) AppleWebKit\/537.36 \(KHTML, like Gecko\)Chrome\/(?<major>\d+)[\d\.]+(?<mobile>[ Mobile]*) Safari\/537\.36$/;
+const matched = chromeUAs.exec(request.get(\'user-agent\'));
```
Comment (agektmr) [inline site/ja/docs/privacy-sandbox/user-agent/snippets/index.md:117]:
```suggestion
const matched = chromeUAs.exec(request.get('user-agent'));
```

File: site/ja/docs/privacy-sandbox/user-agent/snippets/index.md [hunk 17]
```diff
@@ -0,0 +1,170 @@
...
+    'X11; C': 'X11; CrOS x86_64',
+    'X11; L': 'X11; Linux x86_64',
+  };
+  request.headers['user-agent'] = `Mozilla/5.0 (${unifiedPlatform[matched.groups.platform]}) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/${matched.groups.major}.0.0.0${matched.groups.mobile} Safari/537.36`;
+}
+```
+
+### Cloudflare Workers
+
+[Cloudflare Workers](https://developers.cloudflare.com/workers/) を使用すると、サイトへのリクエストをプロキシして変換できます。次のスニペットは、受信した [User-Agent ヘッダー](https://developer.mozilla.org/docs/Web/HTTP/Headers/User-Agent)を情報量削減後の形式に置き換えます。
```
Comment (agektmr) [inline site/ja/docs/privacy-sandbox/user-agent/snippets/index.md:133]:
```suggestion
[Cloudflare Workers](https://developers.cloudflare.com/workers/) を使用すると、サイトへのリクエストをプロキシして変換できます。次のスニペットは、受信した [`User-Agent` ヘッダー](https://developer.mozilla.org/docs/Web/HTTP/Headers/User-Agent)を情報量削減後の形式に置き換えます。
```

File: site/ja/docs/privacy-sandbox/user-agent/snippets/index.md [hunk 18]
```diff
@@ -0,0 +1,170 @@
...
+        unifiedPlatform[matched.groups.platform]
+      }) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/${
+        matched.groups.major
+      }.0.0.0${matched.groups.mobile} Safari/537.36`
+    );
+    return await fetch(clonedRequest);
+  } else {
+    return await fetch(request);
+  }
+}
```
Comment (agektmr) [inline site/ja/docs/privacy-sandbox/user-agent/snippets/index.md:167]: Please replicate the original English version as much as possible (except the Japanese translation).

**Review by agektmr — CHANGES_REQUESTED** [review]

**Comment by mattletford** [comment]
@agektmr I've updated the code examples

File: site/ja/docs/privacy-sandbox/user-agent/snippets/index.md [hunk 19]
```diff
@@ -0,0 +1,181 @@
...
+### 正規表現
+
+確実に変換可能な Chrome の User-Agent **のみ**に一致し、必要な変数値（後で新しい形式のテンプレートに挿入する）を抜き出すための正規表現です。
+
+コピーしやすいように 1 行で記述されています。長いですが、Chrome
+のみに一致させることと比較的低負荷なチェックで済ませることを優先した結果です。
+
+````text
+/^Mozilla\/5\.0 \(((?<platform>Lin|Win|Mac|X11; C|X11; L)+[^\)]+)\) AppleWebKit\/537.36 \(KHTML, like Gecko\) Chrome\/(?<major>\d+)[\d\.]+(?<mobile>[ Mobile]*) Safari\/537\.36$/
+
```
Comment (agektmr) [inline site/ja/docs/privacy-sandbox/user-agent/snippets/index.md:61]: Please append "```" here.

File: site/ja/docs/privacy-sandbox/user-agent/snippets/index.md [hunk 20]
```diff
@@ -0,0 +1,181 @@
...
+情報量削減後の形式の `User-Agent` 文字列を使用できれば、それを自分のコードに照らして変換したりテストしたりできます（オーバーライドや置き換えを行う、新しいバージョンを作成して並列にテストするなど）。
+
+### 正規表現
+
+確実に変換可能な Chrome の User-Agent **のみ**に一致し、必要な変数値（後で新しい形式のテンプレートに挿入する）を抜き出すための正規表現です。
+
+コピーしやすいように 1 行で記述されています。長いですが、Chrome
+のみに一致させることと比較的低負荷なチェックで済ませることを優先した結果です。
+
+````text
```
Comment (agektmr) [inline site/ja/docs/privacy-sandbox/user-agent/snippets/index.md:59]:
```suggestion
```text
```

File: site/ja/docs/privacy-sandbox/user-agent/snippets/index.md [hunk 21]
```diff
@@ -0,0 +1,181 @@
...
+    Mac: 'Macintosh; Intel Mac OS X 10_15_7',
+    'X11; C': 'X11; CrOS x86_64',
+    'X11; L': 'X11; Linux x86_64',
+  };
+  request.headers['user-agent'] = `Mozilla/5.0 (${
+    unifiedPlatform[matched.groups.platform]
+  }) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/${
+    matched.groups.major
+  }.0.0.0${matched.groups.mobile} Safari/537.36`;
+}
```
Comment (agektmr) [inline site/ja/docs/privacy-sandbox/user-agent/snippets/index.md:139]: Please replicate the original English version as much as possible (except the Japanese translation).

File: site/ja/docs/privacy-sandbox/user-agent/snippets/index.md [hunk 22]
```diff
@@ -0,0 +1,181 @@
...
+    `Mozilla/5.0 (${unifiedPlatform[matched.groups.platform]}) ` +
+    `AppleWebKit/537.36 (KHTML, like Gecko) ` +
+    `Chrome/${matched.groups.major}.0.0.0${matched.groups.mobile} Safari/537.36`;
+  // navigator.userAgent を情報量削減後の文字列でオーバーライドする
+  Object.defineProperty(navigator, 'userAgent', {
+    value: reducedUA,
+    writable: false,
+    configurable: true,
+  });
+}
```
Comment (agektmr) [inline site/ja/docs/privacy-sandbox/user-agent/snippets/index.md:112]: Please replicate the original English version as much as possible (except the Japanese translation).

**Review by agektmr — CHANGES_REQUESTED** [review]
The comments are not addressed.

**Comment by mattletford** [comment]
@agektmr I've applied the suggested changes

**Comment by agektmr** [comment]
@mattletford Please read the message, not just apply my suggested changes.

File: site/ja/docs/privacy-sandbox/user-agent/snippets/index.md (hunk 22 above)
Comment (mattletford) [reply site/ja/docs/privacy-sandbox/user-agent/snippets/index.md:113]: > @mattletford Please read the message, not just apply my suggested changes.

I've replicated the original English version as much as possible (except the Japanese translation). If there's anything else I'm missing please let me know.

**Comment by agektmr** [comment]
You are overlooking where my comments belong to. I'm asking to fix lines: 89-112, 136-167 and 122-139.

**Comment by agektmr** [comment]
@mattletford Can you fix them soon? I'll be on long holiday starting tomorrow.

File: site/ja/docs/privacy-sandbox/user-agent/snippets/index.md (hunk 19 above)
Comment (mattletford) [reply site/ja/docs/privacy-sandbox/user-agent/snippets/index.md:56]: I've added "```"

File: site/ja/docs/privacy-sandbox/user-agent/snippets/index.md (hunk 20 above)
Comment (mattletford) [reply site/ja/docs/privacy-sandbox/user-agent/snippets/index.md]: I've removed the backtick

File: site/ja/docs/privacy-sandbox/user-agent/snippets/index.md (hunk 21 above)
Comment (mattletford) [reply site/ja/docs/privacy-sandbox/user-agent/snippets/index.md]: I've replicated this section

File: site/ja/docs/privacy-sandbox/user-agent/snippets/index.md (hunk 22 above)
Comment (mattletford) [reply site/ja/docs/privacy-sandbox/user-agent/snippets/index.md]: I've replicated this section

**Comment by mattletford** [comment]
> You are overlooking where my comments belong to. I'm asking to fix lines: 84-107, 117-134 and 142-173.

I've fixed lines 84-107, 117-134 and 142-173

**Comment by mattletford** [comment]
> @mattletford Can you fix them soon? I'll be on long holiday starting tomorrow.

Yes ofcourse, hopefully this is done now

**Review by agektmr — APPROVED** [review]
LGTM

**Comment by agektmr** [comment]
Thank you, @mattletford !

**Comment by mattletford** [comment]
> Thank you, @mattletford !

You're welcome @agektmr thank you for the help

**Pull request merged**